### ISSUE TITLE
[BUG] Polarizability cannot be correctly trained with `tf` backend

### ISSUE THREAD

**Issue by Yi-FanLi** [deepmodeling/deepmd-kit#4559]
### Bug summary

I am trying to train a `polarizability` model with the example in the GitHub repo. When training the `global`-only polarizability model, I obtain the following learning curve:

![Image](https://github.com/user-attachments/assets/7f72c3bf-277e-4f44-b9a2-98b70e81fa27)

Although the training data seem to be fitted, the trained model is predicting the diagnoal components of polarizability with an **abnormal constant shift**:

![Image](https://github.com/user-attachments/assets/b7f1ef97-af8d-4cab-9c15-08b006fec411)

The non-diagonal components do not show the constant shift:

![Image](https://github.com/user-attachments/assets/e9914db2-9323-458f-b4ae-a459347af2d0)

Moreover, the model cannot work on the **validation dataset** at all:

![Image](https://github.com/user-attachments/assets/29b2b365-0846-41d9-91ff-09e53a1a11ee)

If I extract the diagonal terms of polarizability and fit them with a `dipole` model, **the suspicious constant shift does not appear**:

![Image](https://github.com/user-attachments/assets/cc9806f5-b665-47bd-9d01-e330bb0cdffc)

while the model still does not work at all for the validation data:

![Image](https://github.com/user-attachments/assets/45e86fdc-a36c-4a81-bc51-5b24260bc7ba)

Moreover, the behavior of the `atomic polarizability` model is very similar to the `global polarizability`, i.e., showing the **suspicious constant shift** as well as **failure to generalize on the validation set**:

![Image](https://github.com/user-attachments/assets/b53f5d2a-275e-4e08-a467-fcfa51747562)
![Image](https://github.com/user-attachments/assets/cd956f98-db36-4df4-9728-ad8e63f51125)

### DeePMD-kit Version

DeePMD-kit v3.0.1.dev55+g3e9cf881

### Backend and its version

TensorFlow v2.18.0

### How did you download the software?

pip

### Input Files, Running Commands, Error Log, etc.

`dp train polar_input.json`

### Steps to Reproduce

I used examples in the GitHub repo, but the tasks I used can also be found here:

[polar_issue.tar.gz](https://github.com/user-attachments/files/18459617/polar_issue.tar.gz)

**Comment by Yi-FanLi** [comment]
I want to mention that the shift in the prediction of `polarizability` appears only when `shift_diag=True`. Setting `shift_diag=False` can make accurate fitting of the training data.

**Comment by njzjz** [comment]
Reason: `constant_matrix` is not recovered when the model is recovered. It should be handled in the same way as `bias_atom_e` in the energy model.

**Comment by Yi-FanLi** [comment]
By using PR#4581, the trained model now gives the following output:

![Image](https://github.com/user-attachments/assets/e519e947-26dc-4134-9f4e-a94933c7c5da)

This test result shows that PR#4581 fixes this issue.
